### PR TITLE
Add v1.23 release to release list

### PR DESCRIPTION
**Pull request by sftim**
https://kubernetes.io/releases/ and https://kubernetes.io/releases/patch-releases/ don't list Kubernetes v1.23 at all; I think the should, albeit that the patch releases page would say that there aren't any patch releases yet.

We can also say when v1.23 goes out of support.

[[preview](https://deploy-preview-30800--kubernetes-io-main-staging.netlify.app/releases)]

/sig release
/milestone 1.23

### CODE REVIEW

**Comment by jlbutler** [comment]
It looks like there are regular cadence updates here as part of another mechanism. I’m certainly not opposed to the updates, but I wonder if we’re missing where these are usually added.

**Comment by sftim** [comment]
@jlbutler the `/release` path is fairly recent; I don't think it existed in its current form for the v1.22 release.

Usually folks from SIG Release keep it up to date, but I'm not sure that's covered for minor or major releases.

**Comment by jlbutler** [comment]
I think we should fold this into the release docs workflows, I'll put it on my list. Thanks much @sftim 

What's the source for the EOL info? If we can get that sorted and in place, lgtm.

**Comment by sftim** [comment]
@jlbutler:
> What's the source for the EOL info? If we can get that sorted and in place, lgtm.

I usually look at this page. I don't know where that information stems from; there might be a SIG Docs person who does.
I did mention in SIG Release' Slack channel that it needs filling in. Ultimately I think we have to rely somewhat on SIG Release owning both this section of the website and the information that lives in it.

**Comment by jlbutler** [comment]
removing this from the 1.23 milestone as i need to close it out to wrap 1.23 work. however, this should be handled and is important to resolve.

@cpanato you've done a few updates of this sort in the past. is this something the release team should be doing for each version, or is this typically ok to do later by some other mechanism?

File: content/en/releases/patch-releases.md
Comment (puerco) [inline content/en/releases/patch-releases.md:92]:
```suggestion
**1.23** enters maintenance mode on **2022-12-28**.

End of Life for **1.23** is **2023-02-28**.
```

File: data/releases/schedule.yaml
Comment (puerco) [inline data/releases/schedule.yaml:4]:
```suggestion
  endOfLifeDate: 2023-02-28
```

**Review by puerco — CHANGES_REQUESTED** [review]
These should be the EOL and maintenance mode for Kubernetes 1.23 :point_down:

**Comment by sftim** [comment]
Not sure what Netlify is thinking though. No preview update yet.

**Comment by sftim** [comment]
Local preview

File: content/en/releases/patch-releases.md
Comment (reylejano) [inline content/en/releases/patch-releases.md:94]:
```suggestion
Patch release schedule to be determined.
```
Is it better to state that the 1.23 patch schedule is to be determined to deter an impression of no patch releases for 1.23

**Comment by sftim** [comment]
I also opened https://github.com/kubernetes/website/pull/30924 which makes the `{{< release-data >}}` shortcode render better.

**Pull request merged**
